crates/router/src/connector/stripe.rs

```diff
@@ -1875,7 +1879,11 @@ impl api::IncomingWebhook for Stripe {
                 match details.event_data.event_object.metadata {
                     // if order_id is present
                     Some(meta_data) => api_models::webhooks::ObjectReferenceId::PaymentId(
-                        api_models::payments::PaymentIdType::PaymentAttemptId(meta_data.order_id),
+                        api_models::payments::PaymentIdType::PaymentAttemptId(
```

Comment:
Same here, Instead of checking metadata, extract order_id and match order_id for some or None